src/main/java/duke/command/UserCommand.java

```diff
@@ -0,0 +1,42 @@
+package duke.command;
+
+import duke.exceptions.DukeException;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents the parent class of all commands.
+ */
+public abstract class UserCommand {
+
+    /**
+     * user's input.
+     */
+    protected String userInput;
+
+    /**
+     * boolean to check if user wishes to exit the program
+     */
+    public Boolean isExit = false;
+
+    /**
+     * @param userInput user's input.
+     */
+    public UserCommand(String userInput) {
+        this.userInput = userInput;
+    }
+
+    /**
+     * @param taskList task list containing all the tasks.
+     * @param ui       ui that prints output.
```

Comment:
Should the abundant spaces between the parameter and its description be removed?